diff -urN zaptel-1.4.4.org/tor2.c zaptel-1.4.4/tor2.c
--- zaptel-1.4.4.org/tor2.c	2007-08-02 18:13:39.000000000 +0200
+++ zaptel-1.4.4/tor2.c	2007-11-02 19:15:44.023351857 +0100
@@ -695,7 +695,7 @@
 
 static int __init tor2_init(void) {
 	int res;
-	res = pci_module_init(&tor2_driver);
+	res = zap_pci_module(&tor2_driver);
 	printk("Registered Tormenta2 PCI\n");
 	return res;
 }
diff -urN zaptel-1.4.4.org/wctc4xxp/base.c zaptel-1.4.4/wctc4xxp/base.c
--- zaptel-1.4.4.org/wctc4xxp/base.c	2007-07-02 15:46:37.000000000 +0200
+++ zaptel-1.4.4/wctc4xxp/base.c	2007-11-02 19:15:45.023404110 +0100
@@ -1844,7 +1844,7 @@
 {
 	int res;
 
-	res = pci_module_init(&wcdte_driver);
+	res = zap_pci_module(&wcdte_driver);
 	if (res)
 		return -ENODEV;
 	return 0;
diff -urN zaptel-1.4.4.org/wctdm24xxp.c zaptel-1.4.4/wctdm24xxp.c
--- zaptel-1.4.4.org/wctdm24xxp.c	2007-02-16 23:18:41.000000000 +0100
+++ zaptel-1.4.4/wctdm24xxp.c	2007-11-02 19:15:44.523377825 +0100
@@ -3519,7 +3519,7 @@
 		return -ENODEV;
 	}
 
-	res = pci_module_init(&wctdm_driver);
+	res = zap_pci_module(&wctdm_driver);
 	if (res)
 		return -ENODEV;
 	return 0;
diff -urN zaptel-1.4.4.org/wcte12xp.c zaptel-1.4.4/wcte12xp.c
--- zaptel-1.4.4.org/wcte12xp.c	2007-03-05 06:42:41.000000000 +0100
+++ zaptel-1.4.4/wcte12xp.c	2007-11-02 19:15:44.523377825 +0100
@@ -2012,7 +2012,7 @@
 {
 	int res;
 
-	res = pci_module_init(&te12xp_driver);
+	res = zap_pci_module(&te12xp_driver);
 
 	return res ? -ENODEV : 0;
 }
