diff -urN zaptel-1.4.1.org/Makefile zaptel-1.4.1/Makefile
--- zaptel-1.4.1.org/Makefile	2007-03-23 03:11:12.000000000 +0100
+++ zaptel-1.4.1/Makefile	2007-04-18 23:05:43.346414984 +0200
@@ -170,7 +170,7 @@
 CFLAGS_x86_64:=-m64
 endif
 CFLAGS+=$(CFLAGS_PPC) $(CFLAGS_x86_64)
-KFLAGS=-I$(KINCLUDES) -O6
+KFLAGS=-I$(KINCLUDES) $(OPTFLAGS)
 KFLAGS+=-DMODULE -D__KERNEL__ -DEXPORT_SYMTAB -I$(KSRC)/drivers/net \
 	-Wall -I. -Wstrict-prototypes -fomit-frame-pointer -I$(KSRC)/drivers/net/wan -I$(KINCLUDES)/net
 ifneq (,$(wildcard $(KINCLUDES)/linux/modversions.h))
@@ -293,7 +293,7 @@
 all: menuselect.makeopts 
 	@$(MAKE) _all
 
-_all: $(if $(BUILD_MODULES),modules) programs
+_all: programs
 
 libs: $(LTZ_SO) $(LTZ_A)
 
@@ -427,7 +427,7 @@
 $(UTILSO): %.o: %.c
 	$(CC) $(CFLAGS) -o $@ -c $<
 
-install: all devices install-modules install-programs install-firmware
+install: all devices install-programs install-firmware
 	@echo "###################################################"
 	@echo "###"
 	@echo "### Zaptel installed successfully."
