diff -ur zaptel-1.2.1.org/Makefile zaptel-1.2.1/Makefile
--- zaptel-1.2.1.org/Makefile	2005-12-04 20:29:27.000000000 +0100
+++ zaptel-1.2.1/Makefile	2006-01-12 21:50:07.253302750 +0100
@@ -7,9 +7,8 @@
 
 
 HOSTCC=gcc
-ifeq ($(PWD),)
 PWD:=$(shell pwd)
-endif
+
 # If you want to build for a kernel other than the current kernel, set KVERS
 ifndef KVERS
 KVERS:=$(shell uname -r)
@@ -24,12 +24,12 @@
 endif
 KINCLUDES:=$(KSRC)/include
 
-CFLAGS+=-I. -Iinclude -O4 -g -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
+CFLAGS+=-I. -Iinclude $(OPTFLAGS) -g -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
 CFLAGS_PPC:=$(shell if uname -m | grep -q ppc; then echo "-fsigned-char"; fi)
 CFLAGS_X86-64:=$(shell if uname -m | grep -q x86_64; then echo "-m64"; fi)
 CFLAGS+=$(CFLAGS_PPC) $(CFLAGS_X86-64)
 LCFLAGS:=-fPIC $(CFLAGS) -DBUILDING_TONEZONE
-KFLAGS:=-I$(KINCLUDES) -O6
+KFLAGS:=-I$(KINCLUDES) $(OPTFLAGS)
 KFLAGS+=-DMODULE -D__KERNEL__ -DEXPORT_SYMTAB -I$(KSRC)/drivers/net \
 	-Wall -I. -Wstrict-prototypes -fomit-frame-pointer -I$(KSRC)/drivers/net/wan -I$(KINCLUDES)/net
 ifneq (,$(wildcard $(KINCLUDES)/linux/modversions.h))
@@ -111,7 +111,7 @@
 
 MOD_DESTDIR:=zaptel
 
-EXTRA_CFLAGS:=-I$(PWD)/include  -I$(PWD)/include/oct6100api
+EXTRA_CFLAGS:=-I$(SUBDIRS)/include -I$(SUBDIRS)/include/oct6100api
 obj-m:=$(MODULESO)
 wct4xxp-objs:=wct4xxp_base.o vpm450m.o
 
@@ -177,7 +177,7 @@
 	$(CC) $(KFLAGS) -o $@ -c $<
 
 vpm450m.o: vpm450m.c zaptel.h
-	$(CC) $(KFLAGS) -I$(PWD)/include  -I$(PWD)/include/oct6100api -o $@ -c $<
+	$(CC) $(KFLAGS) -I$(PWD)/include -I$(PWD)/include/oct6100api -I$(SUBDIRS)/include -I$(SUBDIRS)/include/oct6100api -o $@ -c $<
 
 wct4xxp.o: wct4xxp_base.o vpm450m.o
 	$(LD) -r -o $@ wct4xxp_base.o vpm450m.o
@@ -322,20 +322,6 @@
 		install -D -m 755 sethdlc $(INSTALL_PREFIX)/sbin/sethdlc ; \
 	fi
 	if [ -f zttool ]; then install -D -m 755 zttool $(INSTALL_PREFIX)/sbin/zttool; fi
-ifeq ($(BUILDVER),linux26)
-	for x in $(MODULESKO); do \
-		rm -f $(INSTALL_PREFIX)/lib/modules/$(KVERS)/extra/$$x ; \
-	done; \
-	$(KMAKE_INST);
-else
-	for x in $(MODULESO); do \
-		install -D -m 644 $$x $(INSTALL_PREFIX)/lib/modules/$(KVERS)/misc/$$x ; \
-	done;
-endif
-	if ! [ -f wcfxsusb.o ]; then \
-		rm -f $(INSTALL_PREFIX)/lib/modules/$(KVERS)/misc/wcfxsusb.o; \
-	fi; \
-	rm -f $(INSTALL_PREFIX)/lib/modules/$(KVERS)/misc/wcfxs.o
 	install -D -m 755 $(LIBTONEZONE_SO) $(INSTALL_PREFIX)/usr/lib/$(LIBTONEZONE_SO).$(LIBTONEZONE_SO_MAJOR_VER).$(LIBTONEZONE_SO_MINOR_VER)
 	[ `id -u` = 0 ] && /sbin/ldconfig || :
 	rm -f $(INSTALL_PREFIX)/usr/lib/$(LIBTONEZONE_SO)
@@ -349,12 +335,7 @@
 	install -D -m 644 tonezone.h $(INSTALL_PREFIX)/usr/include/tonezone.h
 	install -m 644 doc/ztcfg.8 $(INSTALL_PREFIX)/usr/share/man/man8
 	install -m 644 doc/zttool.8 $(INSTALL_PREFIX)/usr/share/man/man8
-	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 	[ -f $(CONFIG_FILE) ] || install -D -m 644 zaptel.conf.sample $(CONFIG_FILE)
-	build_tools/genmodconf $(BUILDVER) "$(ROOT_PREFIX)" "$(filter-out zaptel,$(MODULES))"
-	@if [ -d /etc/modutils ]; then \
-		/sbin/update-modules ; \
-	fi
 
 install-udev: devices
 
