diff -urN zaptel-1.4.4.org/tor2-hw.h zaptel-1.4.4/tor2-hw.h
--- zaptel-1.4.4.org/tor2-hw.h	2007-08-02 16:13:39.000000000 +0000
+++ zaptel-1.4.4/tor2-hw.h	2007-09-18 09:34:18.220204503 +0000
@@ -124,6 +124,7 @@
 #endif /* __KERNEL__ */
 {
 	{ PCI_VENDOR_ID_PLX, 0x3000, PCI_ANY_ID, PCI_ANY_ID, 0, 0, (unsigned long)"Tormenta 2 Quad E1/PRA - TE400P" }, /* Tormenta 2 */
+	{ PCI_VENDOR_ID_PLX, 0x4000, PCI_ANY_ID, PCI_ANY_ID, 0, 0, (unsigned long)"Tormenta 2 Quad E1/PRA - TE400P" }, /* Tormenta 2 */
 	{ 0, }
 };
 
